google/resource_compute_instance.go

```diff
@@ -969,11 +981,64 @@ func resourceComputeInstanceRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("name", instance.Name)
 	d.Set("description", instance.Description)
 	d.Set("hostname", instance.Hostname)
+	err = waitUntilInstanceHaveTheExpectedStatus(config, d)
+	if err != nil {
+		return fmt.Errorf("Error waiting for status: %s", err)
+	}
+	d.Set("status", instance.Status)
 	d.SetId(instance.Name)
 
 	return nil
 }
 
+func getAllStatusExcept(exceptedStatus string) []string {
+	allStatus := []string{
+		"PROVISIONING",
+		"REPAIRING",
+		"RUNNING",
+		"STAGING",
+		"STOPPED",
+		"STOPPING",
+		"SUSPENDED",
+		"SUSPENDING",
+		"TERMINATED",
+	}
+	for i, status := range allStatus {
+		if exceptedStatus == status {
+			return append(allStatus[:i], allStatus[i+1:]...)
+		}
+	}
+	return nil
+}
+
+func waitUntilInstanceHaveTheExpectedStatus(config *Config, d *schema.ResourceData) error {
```

Comment:
This function breaks pre-existing user configs for an instance that already exists and is `TERMINATED`. `d.Get("status")` will return the schema default of `"RUNNING"` (because the user hasn't set anything for that field) and the function will wait for the `TERMINATED` instance to transition to `RUNNING` which will of course never happen (without outside action).

Reply:
You are right, I did not consider that case. I have updated the code to use the `waitUntilInstanceHaveTheExpectedStatus` method in `resourceComputeInstanceCreate` function instead of `resourceComputeInstanceRead`. Therefore, existing instances (with default `RUNNING` status if not specified) will not be affected. Integration tests I made are still passing.

Reply:
@tysen I have also added an acceptance test to cover that case (`TestAccComputeInstance_statusTerminatedManuallyAndRestart`).